### PR TITLE
Profile pic upload backend

### PR DESCRIPTION
**Pull request by tsumori91**
### What this PR does (required):
- Incomplete draft of profile Image upload on profile edit page.

### Screenshots/Video
https://www.loom.com/share/fb59446dd4e540f383e1c65669dbd1f5

### Bugs
- It uploads and saves the URL locally rather than to a profile. 
- Still need to connect it to a profile based on userID and get the URL from there.
- Need to Delete url from Profile too rather than locally.

### CODE REVIEW

File: client/src/helpers/APICalls/uploadPic.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { AuthApiData } from '../../interface/AuthApiData';
+import { FetchFormOptions } from '../../interface/FetchOptions';
+
+const uploadPictureRequest = async (file: FormData): Promise<AuthApiData> => {
+  const fetchOptions: FetchFormOptions = {
+    method: 'POST',
+    body: file,
+  };
+  console.log(file);
```
Comment (rajivtitus) [inline client/src/helpers/APICalls/uploadPic.ts:9]: remove

File: server/controllers/profilePhoto.js [hunk 2]
```diff
@@ -1,13 +1,40 @@
...
   const file = req.file;
+  try {
+    const fileStr = req.body.pics;
+    const uploadedResponse = await cloudinary.uploader.upload(fileStr, {
+      upload_preset: "dev_setups",
+    });
+    const profile = await Profile.findById(req.id);
+    res.json(uploadedResponse.url);
+  } catch (error) {
+    console.log(error);
```
Comment (rajivtitus) [inline server/controllers/profilePhoto.js:19]: remove

File: server/controllers/profilePhoto.js [hunk 3]
```diff
@@ -1,13 +1,40 @@
...
-exports.deletePic = asyncHandler(async (req, res, next) => {});
+exports.deletePic = asyncHandler(async (req, res, next) => {
+  const id = req.params.id;
+  const profile = Profile.findById(id);
+  if (!profile) {
+    res.status(404);
+    throw new Error("Cannot find profile.");
+  }
+  try {
+    Profile.findByIdAndUpdate(id, { profilePic: "" });
```
Comment (rajivtitus) [inline server/controllers/profilePhoto.js:34]: In the Profile model we only have a profilePhoto field, please update accordingly

**Review by rajivtitus — APPROVED** [review]
Nice work so far! Just a couple of comments.

File: server/controllers/profilePhoto.js (hunk 3 above)
Comment (tsumori91) [reply server/controllers/profilePhoto.js:34]: Very nice catch!

**Pull request merged**